contributing: Make release procedure cleaner and more detailed (#2346)
OSGeo/grass: doc/howto_release.md

## Assumptions

- You have communicated with other developers, e.g., through grass-dev mailing list.
- You have communicated with a development coordinator.
- You have already cloned the repo with Git.
- Your own fork is the remote called "origin".
- The OSGeo repo is the remote called "upstream".
- You don't have any local un-pushed or un-committed changes.
- You are using Bash or a similar shell.

*Note: Some later steps in this text are to be done by the development coordinator
(currently Markus Neteler and Martin Landa) due to needed logins.*

## Prepare the local repo

Update your remotes and switch to branch:

```bash
git fetch --all --prune && git checkout releasebranch_8_2
```

Confirm that you are on the right branch and have no local changes:

```bash
# Should show no changes:
git status
# Should give no output at all:
git diff
git diff --staged
git log upstream/releasebranch_8_2..HEAD
# Should give the same as last commits visible on GitHub:
git log --max-count=5
```

Now you can merge (or rebase) updates from the remote your local branch
and optionally update your own fork:

```bash
git merge upstream/releasebranch_8_2 && git push origin releasebranch_8_2
```

> Use _git log_ and _git show_ to verify the result:

Reviewer: ```suggestion
Use `git log` and `git show` to verify the result:
```

Author: Changed the emphasis with underscores to code with backticks in 01336aa155, but it is actually questionable. While the code block which immediately follows is code to be executed, the names in the text refer to the tools rather than representing the code to be executed and in that context emphasis (italics) is commonly used these days I think. Both however happen to be meaningful code as well.

```bash
git log --max-count=5
git show
```

> Any time later, you can use _git log_ and _git show_ to see the latest

Reviewer: ```suggestion
Any time later, you can use  `git log` and `git show` to see the latest
```

Author: Also changed in 01336aa155.

git push upstream
```

## Create variables

For convenience, create Bash variables with the version update script:

```bash
# Get VERSION and TAG as variables.
eval $(./update_version.py status --bash)
```

Version and tag are the same for all releases:

```bash
echo "$VERSION"
echo "$TAG"
```

> If in doubt, run it without `eval $(...)` to see all the variables created.

Reviewer: ```suggestion
If in doubt, run `update_version.py` without `eval $(...)` to see all the variables created.
```

Author: The path to it was actually missing. Now fixed in d425132ed3 and the text says _the script_.

vim wingrass-maintenance-scripts/cronjob.sh       # major/minor release only
```

## Update addon builder

- https://github.com/landam/wingrass-maintenance-scripts/blob/master/grass_addons.sh (add new release related line)

## Close milestone

> For a (final) release (not release candidate), close the related milestone at

Reviewer: Maybe add a hint what to do with issues/PR yet assigned to this milestone (individually bump to next milestone by editing)?

Author: You can actually move them all at once :-)

Author: In d7523ac40c, I actually added evaluating the state of the milestone to the assumptions as well.
